static/app/components/smartSearchBar/index.tsx

```diff
@@ -565,6 +566,37 @@ class SmartSearchBar extends React.Component<Props, State> {
       this.doSearch();
       return;
     }
+
+    const cursorToken = this.cursorToken;
+    if (
+      key === '[' &&
+      cursorToken?.type === Token.Filter &&
+      cursorToken.value.text.length === 0
+    ) {
+      const {query} = this.state;
+      if (isWithinToken(cursorToken.value, this.cursorPosition)) {
+        evt.preventDefault();
+        let clauseStart: null | number = null;
+        let clauseEnd: null | number = null;
+        // the new text that will exist between clauseStart and clauseEnd
+        const replaceToken = '[]';
+        const location = cursorToken.value.location;
+        const keyLocation = cursorToken.key.location;
+        // Include everything after the ':'
+        clauseStart = keyLocation.end.offset + 1;
+        clauseEnd = location.end.offset + 1;
+        const beforeClause = query.substring(0, clauseStart);
+        let endClause = query.substring(clauseEnd);
+        // Add space before next clause if it exists
+        if (endClause) {
+          endClause = ` ${endClause}`;
+        }
+        const newQuery = `${beforeClause}${replaceToken}${endClause}`;
+        // Place cursor between inserted brackets
+        this.updateQuery(newQuery, beforeClause.length + replaceToken.length - 1);
```

Comment:
We should maybe make a helper function for replacing text in a query. What do you think @davidenwang

Reply:
yea that makes sense will remove the need to keep reconstructing the query